### ISSUE TITLE
SelectCheckboxMenu: add accessibility to items checkbox and label

### ISSUE THREAD

**Issue by Eehol** [primefaces/primefaces#4683]
Hello,

For accessibility purpose (see WCAG2.1 rules), is it possible to add the "for" attribute to the selectCheckboxMenu's labels to reference the left checkbox ?

POC (left=PF 7.0 code, right=POC code) : 
https://i.postimg.cc/qBhpHdTv/primefaces-Select-Checkbox.png

With this change, a screenreader reads "Other - checkbox not checked"
With PF 7.0, it reads "checkbox not checked" without the label, which is not accessible.

NB: possibly linked to : https://github.com/primefaces/primefaces/issues/3019 but screenreaders still doesn't read correctly the component's labels.

**Comment by melloware** [comment]
I am looking at the source for 7.0 and it does add the for="" attribute. See this line: https://github.com/primefaces/primefaces/blob/master/src/main/java/org/primefaces/component/selectcheckboxmenu/SelectCheckboxMenuRenderer.java#L191

**Comment by Eehol** [comment]
Indeed the code you pointed seems correct to me but you can check in the Primefaces showcase that the generated code have no "for" attribute :-/
https://i.postimg.cc/L6Ddqjrh/1553523864277-primefaces-Select-Checkbox.png

**Comment by melloware** [comment]
You are right. I found it for the Multiple style if does it all in JavaScript Side here: https://github.com/primefaces/primefaces/blob/master/src/main/resources/META-INF/resources/primefaces/forms/forms.selectcheckboxmenu.js#L196-L207

**Comment by melloware** [comment]
Submitted a PR.  When testing my PR it now generates the HTML like this...

```xml
<li class="ui-selectcheckboxmenu-item ui-selectcheckboxmenu-list-item ui-corner-all ui-selectcheckboxmenu-checked" data-item-value="London">
   <div class="ui-chkbox ui-widget"><div class="ui-helper-hidden-accessible">
      <input type="checkbox" role="checkbox" readonly="readonly" aria-checked="true" id="142c4b84-4a50-42a3-93d7-e9edbbd11390">
   </div>
   <div class="ui-chkbox-box ui-widget ui-corner-all ui-state-default ui-state-active"><span class="ui-chkbox-icon ui-icon ui-icon-check"></span></div></div>
   <label for="142c4b84-4a50-42a3-93d7-e9edbbd11390">London</label>
</li>
```
